service/k8sclient/k8client.go

```diff
@@ -583,6 +625,32 @@ func (c *K8Client) getPSMDBClusters(ctx context.Context) ([]PSMDBCluster, error)
 	return res, nil
 }
 
+/*
+  When a cluster is being initialized but there are not enough nodes to form a cluster (less than 3)
+  the operator returns State=Error but that's not the real cluster state.
+  While the cluster is being initialized, we need to return the lowest state value found in the
+  replicaset list of members.
+*/
+func getReplicasetStatus(cluster perconaServerMongoDB) ClusterState {
+	if strings.ToLower(string(cluster.Status.Status)) != "error" {
+		return psmdbStatesMap[cluster.Status.Status]
+	}
+
+	if len(cluster.Status.Replsets) == 0 {
+		return ClusterStateInvalid
+	}
+
+	status := ClusterState(99)
```

Comment:
maybe we can use some state from our states list?

Reply:
We need a high high value so any lower value will be chosen instead of this one. If I use the current higher value, lets say `4`, if tomorrow someone adds a `5` value, we must remember to change it here too. That's why I used 99.

Reply:
why do we check for the lowest value?

Reply:
Anyway, I changed it to take the first value as the initial comparison value.